playbooks/remove_node.yml

```diff
@@ -1,9 +1,17 @@
 ---
+- name: Validate nodes before proceeding
+  hosts: localhost
+  tasks:
+    - name: Fail if no nodes are specified for removal
+      fail:
+        msg: "No nodes specified for removal. The `node` variable must be set to the nodes that need removal."
+      when: node is not defined or node | length == 0
+
 - name: Common tasks for every playbooks
   import_playbook: boilerplate.yml
 
 - name: Confirm node removal
-  hosts: "{{ node | default('etcd:k8s_cluster:calico_rr') }}"
+  hosts: "{{ node | default([]) }}"
```

Comment:
IMO you could just remove the defaulting altogether since you're checking the node variable before.